test/integration/json.js

```diff
@@ -52,6 +52,20 @@ module.exports = function(bookshelf) {
         });
     });
 
+    it('can update attributes without affecting _previousAttributes', function() {
```

Comment:
I'm not sure this description fits with what is actually being tested here. It should be the opposite since `previousAttributes` will now detect changes to nested objects.